content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,394 @@
+
+Sender MUST send encoded and compresses proto message in the body of an HTTP POST request and send it to the Receiver via HTTP at a provided URL path. The Receiver MAY specify any HTTP URL path to receive metrics.
+
+Sender MUST send the following "reserved" headers with the HTTP request:
+
+* `Content-Encoding: <compression>`
+
+  Content encoding request header MUST follow [the RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-encoding). Sender MUST use the `snappy` value. More compression algorithms might come in 2.x or beyond.
+
+* `Content-Type: application/x-protobuf` or `Content-Type: application/x-protobuf;proto=<fully qualified name>`
+  
+  <!---
+  TODO(bwplotka): The framing here is a bit inconsistent with deprecation policy. We can either mention sender MUST use either 1.0 or 2.0 message. Or we can say sender MUST use 2.0 message and it MAY/SHOULD use 1.0 message against 1.x receiver. What do we prefer?
+  -->
+  Content type request header MUST follow [the RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html#name-content-type). Sender MUST use `application/x-protobuf` as the only media type. Sender MAY add `;proto=` parameter to the header's value to indicate the fully qualified name of the protobuf message (schema) that was used, from the two mentioned above. As a result, Sender MUST send any of the three supported header values:
+
+  For the message introduced in PRW 1.0, identified by `prometheus.WriteRequest`:
+    * `Content-Type: application/x-protobuf`
+    * `Content-Type: application/x-protobuf;proto=prometheus.WriteRequest`
+  For the message introduced in PRW 2.0, identified by `io.prometheus.write.v2.Request`:
+    * `Content-Type: application/x-protobuf;proto=io.prometheus.write.v2.Request`
+  
+  Sender SHOULD use `Content-Type: application/x-protobuf`, for backward compatibility, when talking to 1.x Receiver. Sender SHOULD use `Content-Type: application/x-protobuf;proto=io.prometheus.write.v2.Request` when talking to Receiver supporting 2.x. More proto messages might come in 2.x or beyond.
+
+* `User-Agent: <name & version of the sender>`
+* `X-Prometheus-Remote-Write-Version: <remote write specificiation version the sender follows>`
+
+  Sender SHOULD use `X-Prometheus-Remote-Write-Version: 0.1.0` for backward compatibility, when using 1.0 proto message.
+
+Sender MAY allow users to send custom HTTP headers; they MUST NOT allow users to configure them in such a way as to send reserved headers.
+
+<!---
+TODO(bwplotka): Kind of repeated statement, feedback welcome how to structure it.
+-->
+The remote write request in the body of the HTTP POST MUST be compressed with [Google’s Snappy](https://github.com/google/snappy). The block format MUST be used -- the framed format MUST NOT be used. The remote write request MUST be encoded using Google Protobuf 3, and MUST use either of the schemas defined above. 
```

Comment:
we could include ` #### compression` sub section with notes about snappy (and other compressions in the future potentially) and how to use them if they have configuration options? `The block format MUST be used -- the framed format MUST NOT be used.` seems worth keeping at least, which would be relevant if within the Go community more and more people start using https://github.com/klauspost/compress/tree/master/s2#s2-compression instead of the default snappy library.